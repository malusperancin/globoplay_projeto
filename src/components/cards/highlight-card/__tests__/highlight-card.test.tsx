/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable testing-library/no-node-access */
import { render } from "@testing-library/react";
import HighlightCardType from "../../highlight-card/highlight-card-type";
import HighlightCard from "../highlight-card";

describe("HighlightCard", () => {
  const card: HighlightCardType = {
    hasVideo: true,
    backgroundUrl: "teste",
    logoUrl: "teste",
    description: "teste",
    titleType: "teste",
    mediaId: true,
    titleId: "teste",
    type: "teste",
    broadcast: true,
    headline: "teste",
    headlineImage: "teste",
    serviceId: 20,
    idSalePage: "teste",
    position: 20,
  };

  const highlightComponent = <HighlightCard data={card} isFocus={true} />;

  const highlightComponentNotFocus = (
    <HighlightCard data={card} isFocus={false} />
  );

  it("should render without throwing an error", () => {
    const { container } = render(highlightComponent);

    expect(container).toMatchSnapshot();
    expect(container).toBeTruthy();
  });

  it("should be focus", () => {
    const { getByTestId } = render(highlightComponent);
    const focus =
      getByTestId("highlight-card").getElementsByClassName("buttonFocus")[0];

    expect(focus).toBeTruthy();
  });

  it("should not be focus", () => {
    const { getByTestId } = render(highlightComponentNotFocus);
    const focus =
      getByTestId("highlight-card").getElementsByClassName("button")[0];

    expect(focus).toBeTruthy();
  });
});
