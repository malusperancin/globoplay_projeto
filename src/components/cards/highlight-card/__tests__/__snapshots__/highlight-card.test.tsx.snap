// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightCard should render without throwing an error 1`] = `
<div>
  <div
    class="container"
    data-testid="highlight-card"
  >
    <button
      class="buttonFocus"
      type="button"
    />
  </div>
</div>
`;
